File: src/test/java/j/examples/rules/guarding/AssertionUtils.java
package j.examples.rules.guarding;

import j.examples.validation.ValidationException;
import org.assertj.core.api.ThrowableAssert;

import static org.assertj.core.api.Assertions.assertThatCode;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

public class AssertionUtils {
    private AssertionUtils() {}

    static void assertValid(ThrowableAssert.ThrowingCallable throwingCallable) {
        assertThatCode(throwingCallable).doesNotThrowAnyException();
    }

    static void assertNotValid(ThrowableAssert.ThrowingCallable throwingCallable, String message) {
        assertThatThrownBy(throwingCallable)
            .isInstanceOf(ValidationException.class)
            .hasMessage(message);
    }
}

File: src/test/kotlin/k/examples/rules/guarding/AssertionUtils.kt
package k.examples.rules.guarding

import k.examples.validation.ValidationException
import org.assertj.core.api.Assertions
import org.assertj.core.api.ThrowableAssert

object AssertionUtils {

    fun assertValid(throwingCallable: ThrowableAssert.ThrowingCallable) {
        Assertions.assertThatCode(throwingCallable).doesNotThrowAnyException()
    }

    fun assertNotValid(throwingCallable: ThrowableAssert.ThrowingCallable, message: String) {
        Assertions.assertThatThrownBy(throwingCallable)
            .isInstanceOf(ValidationException::class.java)
            .hasMessage(message)
    }
}